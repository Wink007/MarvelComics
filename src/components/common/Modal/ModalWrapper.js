/**
 * Use this component to
 * - handle modal closing event
 * - stopPropagation on the div that covers { children }
 * */

// Core
import React, { Component } from 'react';
import { func, node, bool } from 'prop-types';

// Custom
import detectMobile from 'utils/ContextProviderHelpers/detectMobile';

// Styles
import './styles.scss';


class ModalWrapper extends Component {
  static propTypes = {
    children: node,
    onClick: func.isRequired,
    background: bool,
  };

  static defaultProps = {
    background: false,
    children: null,
  };

  stopPropagation = (event) => {
    event.stopPropagation();
  };

  render() {
    const { children, onClick, background } = this.props;
    const deviceDetect = detectMobile();
    const { isMobile } = deviceDetect;
    return (
      <div // eslint-disable-line
        className="ModalWrapper"
        onClick={onClick}
        data-background={background ? 'true' : ''}
        data-device={isMobile ? 'mobile' : 'desktop'}
      >
        <div // eslint-disable-line
          onClick={this.stopPropagation}
        >
          {children}
        </div>
      </div>
    );
  }
}

export default ModalWrapper;
