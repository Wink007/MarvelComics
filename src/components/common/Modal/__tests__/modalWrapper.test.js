import PageObject from 'utils/test/pageObject';
import ModalWrapper from '../ModalWrapper';

const props = {
  background: false,
  children: null,
  onClick: jest.fn(),
};

jest.mock('utils/ContextProviderHelpers/detectMobile');
const mobileDetect = require('utils/ContextProviderHelpers/detectMobile').default;

mobileDetect.mockReturnValue({
  isMobile: false,
  isPhone: false,
  os: true,
});

const page = (new PageObject(props)).setComponent(ModalWrapper, false);
const wrapper = page.getShallowRendered();
let pageTwo;
let wrapperTwo;

describe('Check if the <ModalWrapper /> component rendered',
  () => {
    it('+++ render correctly component', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('+++ check onClick', () => {
      wrapper.find('.ModalWrapper').simulate('click');
      expect(props.onClick).toHaveBeenCalled();
    });

    it('+++ check attribute if background: true', () => {
      wrapper.setProps({
        background: true,
      });
      expect(wrapper.find('.ModalWrapper').props()).toHaveProperty(
        'data-background',
        'true',
      );
    });

    it('+++ check data-device attr for desktop', () => {
      expect(wrapper.find('.ModalWrapper').prop('data-device')).toEqual('desktop');
      wrapper.unmount();
      mobileDetect.mockClear();
    });

    it('+++ check data-device attr for mobile', () => {
      mobileDetect.mockReturnValue({
        isMobile: true,
        isPhone: true,
        os: false,
      });
      pageTwo = (new PageObject(props)).setComponent(ModalWrapper, false);
      wrapperTwo = pageTwo.getShallowRendered();
      expect(wrapperTwo.find('.ModalWrapper').prop('data-device')).toEqual('mobile');
    });
  });
