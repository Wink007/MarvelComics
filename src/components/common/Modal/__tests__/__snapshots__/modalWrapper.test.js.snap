// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the <ModalWrapper /> component rendered +++ render correctly component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalWrapper
    background={false}
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        onClick={[Function]}
      />,
      "className": "ModalWrapper",
      "data-background": "",
      "data-device": "desktop",
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          onClick={[Function]}
        />,
        "className": "ModalWrapper",
        "data-background": "",
        "data-device": "desktop",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
